saga/sagaState.go

```diff
@@ -5,28 +5,28 @@ import (
 	"fmt"
 )
 
+type flag byte
```

Comment:
Sorry, I meant 'or', not 'and'.

If you see an ENDCOMP, you know you've seen a STARTCOMP. If you see an END, you know you've seen a START.

So I guess I'm saying I thought you could summarize the state of a Task with one state, instead of all of the messages you've seen. Am I misunderstanding?